symengine/tests/basic/test_functions.cpp

```diff
@@ -2211,6 +2581,27 @@ TEST_CASE("Acosh: functions", "[functions]")
     r1 = acosh(mul(i2, y))->diff(y);
     r2 = div(i2, sqrt(add(mul(i4, pow(y, i2)), im1)));
     REQUIRE(eq(*r1, *r2));
+
+    r1 = acosh(real_double(2.0));
+    REQUIRE(is_a<RealDouble>(*r1));
+    REQUIRE(std::abs(down_cast<const RealDouble &>(*r1).i - 1.31695789692482)
+            < 1e-12);
+
+    r1 = acosh(complex_double(std::complex<double>(1, 2)));
+    r2 = acosh(real_double(-1.0));
+    REQUIRE(is_a<ComplexDouble>(*r1));
+    REQUIRE(is_a<ComplexDouble>(*r2));
+    REQUIRE(std::abs(std::abs(down_cast<const ComplexDouble &>(*r1).i)
+                     - 1.90908861124732)
+            < 1e-12);
+    REQUIRE(std::abs(std::abs(down_cast<const ComplexDouble &>(*r2).i)
+                     - 3.14159265358979)
+            < 1e-12);
```

Comment:
This test fails because of errors in `acosh` in the libm implementation. You can increase the tolerance level to `1e-10`